src/recursion_guard.rs

```diff
@@ -4,7 +4,7 @@ use ahash::AHashSet;
 /// It's used in `validators/definition` to detect when a reference is reused within itself.
 #[derive(Debug, Clone, Default)]
 pub struct RecursionGuard {
-    ids: Option<AHashSet<usize>>,
+    ids: Option<AHashSet<(usize, usize)>>, // the first one is the id() of the python object, the second of the def ref
```

Comment:
This solves a bug where if you had a `def-ref` -> (something) -> `def-ref` and that something did not change the object (like a function validator from a root model that does nothing in Python mode) then you get a false positive recursion error.

Reply:
`(usize, usize)` look a little opaque - do you think it's worth having a struct here just to document/be clearer?